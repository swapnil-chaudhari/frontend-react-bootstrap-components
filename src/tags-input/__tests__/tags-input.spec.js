import React from 'react';
import renderShallow from 'render-shallow';
import TagsInput from '../tags-input';
import noop from 'src/utils/noop';

describe('<TagsInput>', () => {
    describe('when it is rendered with required props', () => {
        const props = {
            onChange: noop,
            value: []
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<TagsInput { ...props } />).output;
        });

        test('has expected markup', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when it is rendered with quality assurance id', () => {
        const props = {
            onChange: noop,
            value: [],
            qaid: 'TagsInput-QA-wrapper'
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<TagsInput { ...props } />).output;
        });

        test('has expected markup', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when it is disabled', () => {
        const props = {
            disabled: true,
            onChange: noop,
            value: []
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<TagsInput { ...props } />).output;
        });

        test('has expected markup', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
