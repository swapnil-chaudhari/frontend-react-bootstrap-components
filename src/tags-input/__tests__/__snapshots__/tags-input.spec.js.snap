// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagsInput> when it is disabled has expected markup 1`] = `
<div
  id={undefined}
>
  <TagsInput
    addKeys={
      Array [
        9,
        13,
      ]
    }
    addOnBlur={false}
    addOnPaste={false}
    className="form-control TagsInput TagsInput-disabled"
    disabled={true}
    focusedClassName="TagsInput-focused"
    inputProps={
      Object {
        "className": "TagsInput-input",
        "placeholder": "",
      }
    }
    maxTags={-1}
    onChange={[Function]}
    onlyUnique={false}
    pasteSplit={[Function]}
    preventSubmit={true}
    removeKeys={
      Array [
        8,
      ]
    }
    renderInput={[Function]}
    renderLayout={[Function]}
    renderTag={[Function]}
    tagDisplayProp={null}
    tagProps={
      Object {
        "className": "TagsInput-tag",
        "classNameRemove": "TagsInput-remove",
      }
    }
    validationRegex={/\\.\\*/}
    value={Array []}
  />
</div>
`;

exports[`<TagsInput> when it is rendered with quality assurance id has expected markup 1`] = `
<div
  id="TagsInput-QA-wrapper"
>
  <TagsInput
    addKeys={
      Array [
        9,
        13,
      ]
    }
    addOnBlur={false}
    addOnPaste={false}
    className="form-control TagsInput"
    disabled={false}
    focusedClassName="TagsInput-focused"
    inputProps={
      Object {
        "className": "TagsInput-input",
        "placeholder": "",
      }
    }
    maxTags={-1}
    onChange={[Function]}
    onlyUnique={false}
    pasteSplit={[Function]}
    preventSubmit={true}
    removeKeys={
      Array [
        8,
      ]
    }
    renderInput={[Function]}
    renderLayout={[Function]}
    renderTag={[Function]}
    tagDisplayProp={null}
    tagProps={
      Object {
        "className": "TagsInput-tag",
        "classNameRemove": "TagsInput-remove",
      }
    }
    validationRegex={/\\.\\*/}
    value={Array []}
  />
</div>
`;

exports[`<TagsInput> when it is rendered with required props has expected markup 1`] = `
<div
  id={undefined}
>
  <TagsInput
    addKeys={
      Array [
        9,
        13,
      ]
    }
    addOnBlur={false}
    addOnPaste={false}
    className="form-control TagsInput"
    disabled={false}
    focusedClassName="TagsInput-focused"
    inputProps={
      Object {
        "className": "TagsInput-input",
        "placeholder": "",
      }
    }
    maxTags={-1}
    onChange={[Function]}
    onlyUnique={false}
    pasteSplit={[Function]}
    preventSubmit={true}
    removeKeys={
      Array [
        8,
      ]
    }
    renderInput={[Function]}
    renderLayout={[Function]}
    renderTag={[Function]}
    tagDisplayProp={null}
    tagProps={
      Object {
        "className": "TagsInput-tag",
        "classNameRemove": "TagsInput-remove",
      }
    }
    validationRegex={/\\.\\*/}
    value={Array []}
  />
</div>
`;
